File: client/lib/models/responses/today_plan_response.dart
import 'workout_template.dart';

/// `GET /api/training/today` payload — both fields independently nullable
/// because the four combinations are all real UX states:
///   * `template == null && activeSessionId == null` → rest day
///   * `template != null && activeSessionId == null` → ready to start
///   * `template == null && activeSessionId != null` → yesterday's session
///     still open
///   * `template != null && activeSessionId != null` → today's plan +
///     in-progress session (offer Resume, not Start).
class TodayPlanResponse {
  TodayPlanResponse({this.template, this.activeSessionId});

  final WorkoutTemplate? template;
  final int? activeSessionId;

  bool get hasPlan => template != null;
  bool get hasActiveSession => activeSessionId != null;

  factory TodayPlanResponse.fromJson(Map<String, dynamic> json) {
    final t = json['template'];
    return TodayPlanResponse(
      template: t == null ? null : WorkoutTemplate.fromJson(t as Map<String, dynamic>),
      activeSessionId: (json['activeSessionId'] as num?)?.toInt(),
    );
  }
}

File: api/src/main/java/com/example/athletehub/dto/TodayPlanResponse.java
package com.example.athletehub.dto;

/**
 * {@code GET /api/training/today} payload.
 *
 * <ul>
 *   <li>{@code template} = null → no plan scheduled for today (rest day).</li>
 *   <li>{@code activeSessionId} != null → the user already has an
 *       in-progress session; the client should offer "Resume" instead of
 *       "Start".</li>
 * </ul>
 *
 * <p>Both fields independently nullable because the four combinations are
 * all real states: no plan + no active (rest day), plan + no active (ready
 * to start), no plan + active (yesterday's session still open), plan +
 * active (today's planned, but you're already in one).
 */
public record TodayPlanResponse(
        WorkoutTemplateDto template,
        Long activeSessionId
) {}
